dowhy/causal_estimator.py

```diff
@@ -686,18 +664,18 @@ def signif_results_tostr(self, signif_results):
 
 
 def estimate_effect(
+    data: pd.DataFrame,
     treatment: Union[str, List[str]],
     outcome: Union[str, List[str]],
-    identified_estimand: IdentifiedEstimand,
```

Comment:
why is identified_estimand removed from the params? Logically, a user should provide the output of identify_effect to this function. 
I see that you are using the estimator's target_estimand, but then having a str parameter feels odd. What are the possible values for identifier name?

Rather than identifier_name, I recommend passing the full identifiedestimand object. It might be redundant now, but can be useful when we are using an external Estimator implementation. We need not assume too much about what the estimator contains.

Reply:
What information from the removed parameter are we missing from the estimator target_estimand? shouldn't they be the same? the identifier name replaces the first part of the previous parameter `method_name` which was a string. According to the docs it:

> Currently requires an explicit method name to be specified. Method names follow the convention of identification method followed by the specific estimation method: "[backdoor/iv].estimation_method_name". 


So, the values that it can take are "backdoor" or "iv", which I think can't be found anywhere in the `target_estimand` or the removed `identified_estimand` param, let me know if this is correct, or if we have another way of getting that "backdoor" or "iv" information.

Reply:
Good point. There's another case to consider, where a user will not provide any estimator and the function will automatically find the right Estimator and initialize it. This is the direction we'll be moving in the future. In that case, we would need the `identified_estimand` parameter.

Also, the identifier_name parameter can be removed, I guess. It was useful earlier because we wanted to specify which identifier name each estimator is connected to. But now since the estimator already has the estimand object, we can consider adding that string as an instance attribute for each Estimator. Each estimator corresponds to either of "backdoor"/"iv". And each estimator could check: if the identified_estimand does not have an entry for the estimator's identifier_name, then it can raise an Error (not for this PR, but later). In that case, L723 is not needed, and L731-733 could be moved to init methods of each estimator. What do you think?

So maybe we can keep the identifier_name parameter for this PR, but could be removed in future.